federation/cmd/federation-controller-manager/app/options/options.go

```diff
@@ -32,6 +32,14 @@ type ControllerManagerConfiguration struct {
 	Port int `json:"port"`
 	// address is the IP address to serve on (set to 0.0.0.0 for all interfaces).
 	Address string `json:"address"`
+	// dnsProvider is the provider for dns services.
+	DnsProvider string `json:"dnsProvider"`
+	// dnsConfigFile is the path to the dns provider configuration file.
+	DnsConfigFile string `json:"ndsConfigFile"`
```

Comment:
@quinton-hoole @mfanjie `ndsConfigFile` probably should be `dnsConfigFile`

(still exists on [master](https://github.com/mfanjie/kubernetes/blob/master/federation/cmd/federation-controller-manager/app/options/options.go#L38))